### DISCUSSION TITLE
Binding for @dynamicMemberLookup members

### DISCUSSION THREAD

**Discussion by piterwilson** [kylehickinson/SwiftUI-WebView#10] [General]
Hi @kylehickinson

 I need to be able to pass the members created using `@dynamicMemberLookup` such as `title`, `url` and the rest as a `Binding`.

I wanted to extract away the logic for a custom `View` that reacts to `title` (for example).

I found myself doing this because passing the member directly did not cause the custom `View` to update:

```swift
extension WebViewStore {
    // Creates a `Binding` for any member created using @dynamicMemberLookup
    func binding<Value>(for keyPath: KeyPath<WebViewStore, Value>) -> Binding<Value> {
        Binding(get: { self[keyPath: keyPath] }, set: {_ in })
    }
}

...

struct BrowserTitleBarView: View {
    @Binding var title: String?
    var body: some View {
        HStack {
            Text(verbatim: title ?? "")
        }
    }
}

...
struct BrowserView: View {

    @StateObject var webViewStore = WebViewStore()

    var body: some View {
        VStack {
               BrowserTitleBarView(title: webViewStore.binding(for: \.title))
               ...
               }
       }
```

I am not an expert on SwiftUI. Is there a better way without creating the `extension`?

Thanks.

**Comment by kylehickinson** [comment]
Hey @piterwilson, what you're describing actually exists already in SwiftUI. The API is called [`Binding.constant(_:)`](https://developer.apple.com/documentation/swiftui/binding/constant(_:)) which lets you create a `Binding` out of something that isn't mutable.  This would let you do something like: 

```swift
BrowserTitleBarView(title: .constant(webViewStore.title))
```

 However you don't need to make variables in your View a Binding just to pass info in, you can just do `var title: String?` for instance and just pass in the title directly.  Bindings are usually only used when you want the View to be able to mutate something a parent passes in.

Let me know if that helps!
